pkg/apis/k0s/v1beta1/cplb.go

```diff
@@ -131,12 +152,36 @@ func (k *KeepalivedSpec) validateVRRPInstances(getDefaultNICFn func() (string, e
 		getDefaultNICFn = getDefaultNIC
 	}
 	for i := range k.VRRPInstances {
-		if k.VRRPInstances[i].Interface == "" {
-			nic, err := getDefaultNICFn()
-			if err != nil {
-				errs = append(errs, fmt.Errorf("failed to get default NIC: %w", err))
+		switch k.VRRPInstances[i].InterfaceType {
+		case VRRPInterfaceTypeInterface, VRRPInterfaceTypeDefault:
+			if len(k.VRRPInstances[i].InterfaceMACAddresses) != 0 {
+				errs = append(errs, errors.New("InterfaceMACAddresses must not be defined unless InterfaceType is mac"))
+			}
+			if k.VRRPInstances[i].Interface == "" {
+				iface, err := getDefaultNICFn()
+				if err != nil {
+					errs = append(errs, fmt.Errorf("failed to get default NIC: %w", err))
+				}
+				k.VRRPInstances[i].Interface = iface
+			}
+		case VRRPInterfaceTypeMAC:
+			if len(k.VRRPInstances[i].InterfaceMACAddresses) == 0 {
```

Comment:
We might want to distinguish `nil` from an empty slice. Would be different errors (required vs. invalid).